scenes/globals/game_state/game_state.gd

```diff
@@ -23,27 +27,60 @@ signal collected_items_changed(updated_items: Array[InventoryItem])
 ## when the player returns to Fray's End the loom can trigger a brief cutscene.
 var incorporating_threads: bool = false
 
+var _state := ConfigFile.new()
+
+
+func _ready() -> void:
+	var err := _state.load(GAME_STATE_PATH)
+	if err != OK and err != ERR_FILE_NOT_FOUND:
+		push_error("Failed to load %s: %s" % [GAME_STATE_PATH, err])
+	_restore()
+
 
 ## Reset the [member inventory] when a quest starts.
 func start_quest() -> void:
 	inventory.clear()
+	_update_inventory_state()
+	_save()
 
 
 ## Add the [InventoryItem] to the [member inventory].
 func add_collected_item(item: InventoryItem) -> void:
-	if not item in inventory:
-		inventory.append(item)
+	inventory.append(item)
 	item_collected.emit(item)
 	collected_items_changed.emit(items_collected())
+	_update_inventory_state()
+	_save()
 
 
 ## Remove the [InventoryItem] from the [member inventory].
 func remove_consumed_item(item: InventoryItem) -> void:
 	inventory.erase(item)
 	item_consumed.emit(item)
 	collected_items_changed.emit(items_collected())
+	_update_inventory_state()
+	_save()
 
 
 ## Return all the items collected so far in the [member inventory].
 func items_collected() -> Array[InventoryItem]:
 	return inventory.duplicate()
+
+
+func _update_inventory_state() -> void:
+	var amount: int = clamp(inventory.size(), 0, InventoryItem.ItemType.size())
+	_state.set_value(INVENTORY_SECTION, INVENTORY_ITEMS_AMOUNT_KEY, amount)
+
+
+func _restore() -> void:
+	var amount_in_state: int = _state.get_value(INVENTORY_SECTION, INVENTORY_ITEMS_AMOUNT_KEY, 0)
+	var amount: int = clamp(amount_in_state, 0, InventoryItem.ItemType.size())
+	for index in range(amount):
+		var item := InventoryItem.with_type(index)
```

Comment:
I'm a little worried that it is going to be confusing if the threads are restored in a different order. I guess we can just say that it's a bug to have them in the wrong order in a quest though!

Reply:
Yes. The other option I handled (and forgot to mention) is: Remove the `@export var item: InventoryItem` from the collectible item, and let the game set it according to the current amount of items. That means current scenes that have the collectible item instantiated won't have the final thread color, and learners won't be able to set it. But maybe that's fine?

Reply:
I think what we have here is OK for now. Let's see how it works out in practice.